editor/src/components/canvas/canvas-strategies/resize-absolute-strategy.ts

```diff
@@ -0,0 +1,473 @@
+        propsToSkip.push(propPathToUpdate)
+      } else {
+        const pinIsPercentage = existingProp.value == null ? false : existingProp.value.unit === '%'
+        let valueToUse: string | number
+        if (parentFrame == null) {
+          valueToUse = absoluteValue
+        } else {
+          valueToUse = valueToUseForPin(propToUpdate, absoluteValue, pinIsPercentage, parentFrame)
+        }
+        propsToSet.push({
+          path: propPathToUpdate,
+          value: jsxAttributeValue(valueToUse, emptyComments),
+        })
+      }
+    })
+  }
+
+  return propsToSet
+}
+
+function updateResizeStrategyState(
+  current: AbsoluteResizeStrategyState,
+  startForNewProperties: CanvasPoint,
+  drag: CanvasVector | null,
+  targetProperty: LayoutTargetableProp | undefined,
+  enableSnapping: boolean,
+  centerBasedResize: boolean,
+  keepAspectRatio: boolean,
+): AbsoluteResizeStrategyState {
+  let propertyAlreadyExists: boolean = false
+  let updatedProperties = current.properties.map((prop) => {
+    if (prop.targetProperty === targetProperty) {
+      propertyAlreadyExists = true
+      return updateResizeDragStatePropertyChange(
+        prop,
+        drag,
+        targetProperty,
+        enableSnapping,
+        centerBasedResize,
+        keepAspectRatio,
+      )
+    } else {
+      return prop
+    }
+  })
+  if (!propertyAlreadyExists) {
+    updatedProperties = [
+      ...current.properties,
+      resizeDragStatePropertyChange(
+        startForNewProperties,
+        drag,
+        enableSnapping,
+        centerBasedResize,
+        keepAspectRatio,
+        targetProperty,
+      ),
+    ]
+  }
+  return keepDeepReferenceEqualityIfPossible(current, {
+    ...current,
+    properties: updatedProperties,
+  })
+}
+
+export const resizeAbsoluteStrategy: CanvasStrategy = {
+  name: 'Resize Absolute Elements',
+  fitnessFn: (editor, currentSession) => {
+    if (editor.selectedViews.length === 0) {
+      return null
+    }
+    for (const selectedView of editor.selectedViews) {
+      const isAbsolute = MetadataUtils.isPositionAbsolute(
+        MetadataUtils.findElementByElementPath(editor.jsxMetadata, selectedView),
+      )
+      if (!isAbsolute) {
+        return null
+      }
+    }
+    // More than one selected view which is absolute positioned, so this should work.
+    return 10
+  },
+  updateFn: (
+    lifecycle: 'transient' | 'final',
+    editorState: EditorState,
+    sessionProps: SelectModeCanvasSessionProps,
+    sessionState: SelectModeCanvasSessionState,
+  ): CanvasStrategyUpdateFnResult => {
+    if (sessionProps.activeControl.type === 'RESIZE_HANDLE') {
+      const resizeHandle = sessionProps.activeControl
+
+      const elementsToTarget = determineElementsToOperateOnForDragging(
+        editorState.selectedViews,
+        editorState.jsxMetadata,
+        false,
+        true,
+      )
+
+      const newSize = calculateNewBounds(editorState, resizeHandle, sessionProps, sessionState)
```

Comment:
TODO: the snapping happens inside `calculateDraggedRectangle` inside `calculateNewBounds`, this function should return the snap guidelines and the "winning" snap, so we can feed that to the editor state patch, so that a new snapping control can draw the snaplines based on what we calculated here. 
The benefit of this approach is that we don't need to maintain two parallel places with snapping code, the calculations happen here, and the snapping controls just draw what the editor state patch instructs them to draw.